src/libostree/ostree-repo.c

```diff
@@ -3911,77 +3911,65 @@ ostree_repo_sign_commit (OstreeRepo     *self,
                          GCancellable   *cancellable,
                          GError        **error)
 {
-  gboolean ret = FALSE;
   g_autoptr(GBytes) commit_data = NULL;
   g_autoptr(GBytes) signature = NULL;
-  g_autoptr(GVariant) commit_variant = NULL;
-  g_autoptr(GVariant) old_metadata = NULL;
-  g_autoptr(GVariant) new_metadata = NULL;
-  glnx_unref_object OstreeGpgVerifyResult *result = NULL;
-  GError *local_error = NULL;
 
+  g_autoptr(GVariant) commit_variant = NULL;
   if (!ostree_repo_load_variant (self, OSTREE_OBJECT_TYPE_COMMIT,
                                  commit_checksum, &commit_variant, error))
-    {
-      g_prefix_error (error, "Failed to read commit: ");
-      goto out;
-    }
+    return g_prefix_error (error, "Failed to read commit: "), FALSE;
```

Comment:
Seeing this yet again made me really want to scratch that itch for the fun of it: https://github.com/GNOME/libglnx/pull/45.